src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/IModelMetadataProvider.cs

```diff
@@ -12,4 +13,12 @@ public interface IModelMetadataProvider
 
         IEnumerable<ModelMetadata> GetMetadataForProperties(Type modelType);
     }
+
+    // TODO: Could we take a breaking change and add this to the IModelMetadataProvider
+    // interface? If not, it means that anyone who's built a custom IModelMetadataProvider
+    // will not get the new parameter validation feature.
+    public interface IExtendedModelMetadataProvider : IModelMetadataProvider
```

Comment:
I don't think this we can that in a point release sadly. 

Long term I think the best thing to do is to add an abstract class so these kinds of things aren't breaking changes.

Reply:
And by **this**, I mean the breaking change.

Reply:
OK, I've replaced `IExtendedModelMetadataProvider` with a new abstract base class that itself implements `IModelMetadataProvider`.